desktop/src/com/mygdx/game/client/hud/GameRoomHUD.java

```diff
@@ -204,11 +220,35 @@ public void changed (ChangeEvent event, Actor actor) {
     stage.addActor(mapSizeSelectBox);
   }
 
+  private void removePlayer(String playerName) {
+    connectService.removeUser(playerName);
+  }
+
   private void startGame() {
     gameStartService.startGame(401);
   }
 
-  public void setPlayers(List<PlayerLobby> players) {
-    this.players = players;
+  private void addBot() {
+    String[] args = {};
+
+    var thread = new Thread(() -> BotLauncher.main(args));
+    thread.start();
+  }
+
+  private static void printLines(String name, InputStream ins) throws Exception {
+    String line = null;
+    BufferedReader in = new BufferedReader(
+            new InputStreamReader(ins));
+    while ((line = in.readLine()) != null) {
+      System.out.println(name + " " + line);
+    }
+  }
+
+  private static void runProcess(String command) throws Exception {
+    Process pro = Runtime.getRuntime().exec(command);
+    printLines(command + " stdout:", pro.getInputStream());
+    printLines(command + " stderr:", pro.getErrorStream());
+    pro.waitFor();
+    System.out.println(command + " exitValue() " + pro.exitValue());
```

Comment:
ajj, myślałem, że to dodasz po stronie serwera, żeby on tym zarządzał 

Reply:
i też myślałem, że to będzie jako `jar` na dysku do odpalenia po stronie serwera, ale w sumie to co jest też działa

jakbyś jednak to chciał odpalać po stronie serwera, to pewnie taka dodatkowa zależność, żeby móc odpalać BotLauncher byłaby problematyczna, ale nie wiem - może dlatego zdecydo2wałeś, żeby robić to u klienta

Reply:
nie, nie pomyślałem o tym, spróboję po stronie serwera, może się uda

Reply:
Ogólnie spoko to jest bo mega szybko się odpala jak dodaje przez klienta, w porównaniu do uruchamiania bota osobno